llvm/test/CodeGen/DirectX/UAddc.ll

```diff
@@ -0,0 +1,72 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -scalarizer -dxil-op-lower -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s
+
+; CHECK: %dx.types.i32c = type { i32, i1 }
+
+define noundef i32 @test_UAddc(i32 noundef %a, i32 noundef %b) {
+; CHECK-LABEL: define noundef i32 @test_UAddc(
+; CHECK-SAME: i32 noundef [[A:%.*]], i32 noundef [[B:%.*]]) {
+; CHECK-NEXT:    [[UADDC1:%.*]] = call [[DX_TYPES_I32C:%.*]] @[[DX_OP_BINARYWITHCARRYORBORROW_I32:[a-zA-Z0-9_$\"\\.-]*[a-zA-Z_$\"\\.-][a-zA-Z0-9_$\"\\.-]*]](i32 44, i32 [[A]], i32 [[B]]) #[[ATTR0:[0-9]+]]
```

Comment:
I know this is auto generated but `DX_TYPES_I32C ` and `DX_OP_BINARYWITHCARRYORBORROW_I32` are significantly obscuring  what we are testing

Reply:
Please don't use `update_test_checks` and then edit the output. `update_test_checks` needs to be idempotent to be useful at all. If we really want to edit this output we *must* remove the "Assertions have been autogenerated" note at the top of the file and manually maintain this test forever rather than using the script, and if we wanted to do that we just shouldn't use the script at all.

Reply:
This is my bad. In this case i don’t think we should use update_test_checks at all because it is doing things that make it hard to spot check if we are emitting the right dxil.